include/robot_interfaces/monitored_robot_driver.hpp

```diff
@@ -172,32 +175,45 @@ class MonitoredRobotDriver : public RobotDriver<Action, Observation>
 
         // loop until shutdown and monitor action timing
         for (size_t t = 0; !is_shutdown_; t++)
-        {
-            bool action_has_ended_on_time =
-                action_end_logger_.wait_for_timeindex(t,
-                                                      max_action_duration_s_);
-            if (!action_has_ended_on_time)
-            {
-                std::cout
-                    << "Action did not end on time, shutting down. Any further "
-                       "actions will be ignored."
-                    << std::endl;
-                shutdown();
-                return;
-            }
-
-            bool action_has_started_on_time =
-                action_start_logger_.wait_for_timeindex(
-                    t + 1, max_inter_action_duration_s_);
-            if (!action_has_started_on_time)
-            {
-                std::cout << "Action did not start on time, shutting down. Any "
-                             "further actions will be ignored."
-                          << std::endl;
-                shutdown();
-                return;
-            }
-        }
+	  {
+	    if(! std::isinf(max_action_duration_s_))
```

Comment:
This update is to make the class robust to infinite durations (without these changes it would hang indefinitely on hand). Note that this change is not related to the initial purpose of this pull request. Creating a new pull request sounds too much overhead to me, but will never do it if requested